src/libs/ReportUtils.ts

```diff
@@ -1777,69 +1815,74 @@ function getTransactionReportName(reportAction) {
-        comment,
+        formattedAmount: CurrencyUtils.convertToDisplayString(transactionDetails?.amount ?? 0, transactionDetails?.currency ?? '', TransactionUtils.isDistanceRequest(transaction)) ?? '',
+        comment: transactionDetails?.comment ?? '',
     });
 }
 
 /**
  * Get money request message for an IOU report
  *
- * @param {Object} report
- * @param {Object} [reportAction={}] This can be either a report preview action or the IOU action
- * @param {Boolean} [shouldConsiderReceiptBeingScanned=false]
- * @param {Boolean} isPreviewMessageForParentChatReport
- * @returns  {String}
- */
-function getReportPreviewMessage(report, reportAction = {}, shouldConsiderReceiptBeingScanned = false, isPreviewMessageForParentChatReport = false) {
-    const reportActionMessage = lodashGet(reportAction, 'message[0].html', '');
-
-    if (_.isEmpty(report) || !report.reportID) {
+ * @param [reportAction] This can be either a report preview action or the IOU action
+ */
+function getReportPreviewMessage(
+    report: OnyxEntry<Report>,
+    reportAction?: OnyxEntry<ReportAction>,
+    shouldConsiderReceiptBeingScanned = false,
+    isPreviewMessageForParentChatReport = false,
+): string {
+    const reportActionMessage = reportAction?.message?.[0].html ?? '';
+    if (isEmptyObject(report) || !report?.reportID) {
         // The iouReport is not found locally after SignIn because the OpenApp API won't return iouReports if they're settled
         // As a temporary solution until we know how to solve this the best, we just use the message that returned from BE
         return reportActionMessage;
     }
 
-    if (!isIOUReport(report) && ReportActionsUtils.isSplitBillAction(reportAction)) {
+    if (!isIOUReport(report) && reportAction && ReportActionsUtils.isSplitBillAction(reportAction)) {
         // This covers group chats where the last action is a split bill action
         const linkedTransaction = TransactionUtils.getLinkedTransaction(reportAction);
-        if (_.isEmpty(linkedTransaction)) {
+        if (Object.keys(linkedTransaction ?? {}).length === 0) {
             return reportActionMessage;
         }
-        if (TransactionUtils.isReceiptBeingScanned(linkedTransaction)) {
-            return Localize.translateLocal('iou.receiptScanning');
+
+        if (isNotEmptyObject(linkedTransaction)) {
+            if (TransactionUtils.isReceiptBeingScanned(linkedTransaction)) {
+                return Localize.translateLocal('iou.receiptScanning');
+            }
+
+            const transactionDetails = getTransactionDetails(linkedTransaction);
+            const formattedAmount = CurrencyUtils.convertToDisplayString(transactionDetails?.amount ?? 0, transactionDetails?.currency ?? '');
+            return Localize.translateLocal('iou.didSplitAmount', {formattedAmount, comment: transactionDetails?.comment ?? ''});
         }
```

Comment:
```suggestion
        if (TransactionUtils.isReceiptBeingScanned(linkedTransaction)) {
            return Localize.translateLocal('iou.receiptScanning');
        }

        const transactionDetails = getTransactionDetails(linkedTransaction);
        const formattedAmount = CurrencyUtils.convertToDisplayString(transactionDetails?.amount ?? 0, transactionDetails?.currency ?? '');
        return Localize.translateLocal('iou.didSplitAmount', {formattedAmount, comment: transactionDetails?.comment ?? ''});
```

Reply:
`isNotEmptyObject(linkedTransaction)` is needed here for type narrowing because `linkedTransaction` can be `{}` and we cannot pass. `{}` to `TransactionUtils.isReceiptBeingScanned` and `getTransactionDetails`

Reply:
I think I got around this by doing some changes above

Reply:
Using https://github.com/Expensify/App/pull/29012#discussion_r1390227738, you can do this suggestion too